internal/backends/winit/winitwindowadapter.rs

```diff
@@ -394,7 +394,7 @@ impl WindowAdapterInternal for WinitWindowAdapter {
             // If we're in fullscreen state, don't try to resize the window but maintain the surface
             // size we've been assigned to from the windowing system. Weston/Wayland don't like it
             // when we create a surface that's bigger than the screen due to constraints (#532).
-            if winit_window.fullscreen().is_some() {
+            if winit_window.fullscreen().is_some() || winit_window.is_maximized() {
```

Comment:
We still need to apply the constraints so when we un-maximize, it should be within the constraints.
What exactly is causing the problem.  Is it the setting the constaints or is it adjust_window_size_to_satisfy_constraints ?

Reply:
I think it means that this is broken for full screen.

I believe the code was written with the assumption that the env variable was the only way to set the window to fullscreen.

fullscreen is not a usual state of windows, so the bug was not apparent before.  But with maximized, we should still be able to keep the constraint properly set  (and also for fullscreen, while we are at it)